pages/words/[id].js

```diff
@@ -0,0 +1,94 @@
+import {
+  StyledDefinition,
+  StyledJPDefinition,
+  StyledUl,
+} from "@/components/Entry";
+import { MainContent } from "@/components/StyledComponents/MainContent";
+import { StyledSecondaryButton } from "@/components/StyledComponents/StyledButtons";
+import { StyledCard } from "@/components/StyledComponents/StyledCard";
+import {
+  StyledCenterAlign,
+  StyledSectionLeftAlign,
+} from "@/components/StyledComponents/StyledSection";
+import { getVisualDate } from "@/utils/helperFunctions";
+import { useRouter } from "next/router";
+
+import styled from "styled-components";
+
+export default function WordDetail({ wordList, databaseIsLoading }) {
+  // const [isEditMode, setIsEditMode] = useState(false);
+  const router = useRouter();
+  const { id } = router.query;
+
+  if (databaseIsLoading) {
+    return <p>Loading...</p>;
+  }
+
+  const entryData = wordList.find(({ _id }) => _id === id);
```

Comment:
What if I manually enter an `id` in the address bar that does not exist?

Your app currently throws an error in that case. Maybe you can safeguard your app against invalid ids by checking if the `entryData` is `undefined`? And if that's the case, return some JSX that says something like, "sorry, there's no entry".

Reply:
Yes good point. Thank you :)